packages/host/app/components/operator-mode/interact-submode.gts

```diff
@@ -278,31 +283,43 @@ export default class InteractSubmode extends Component<Signature> {
         await here.args.saveCard(card);
       },
       delete: async (card: CardDef | URL | string): Promise<void> => {
-        let loadedCard: CardDef;
+        if (here.cardToDelete) {
+          return here.delete.perform(here.cardToDelete.id);
+        }
 
-        if (typeof card === 'string') {
-          let _loadedCard = await here.cardService.getCard(card);
-          if (!_loadedCard) {
-            throw new Error(`Could not load card ${card}`);
-          }
-          loadedCard = _loadedCard;
+        let cardToDelete: CardToDelete | undefined;
+
+        if (typeof card === 'object' && 'id' in card) {
+          let loadedCard = card as CardDef;
+          cardToDelete = {
+            id: loadedCard.id,
+            title: loadedCard.title,
+          };
         } else {
-          loadedCard = card as CardDef;
-        }
+          let cardUrl = card instanceof URL ? card : new URL(card as string);
+          try {
+            let loadedCard = await here.cardService.getCard(cardUrl);
+            cardToDelete = {
+              id: loadedCard.id,
+              title: loadedCard.title,
+            };
+          } catch (error: any) {
+            // Case when card is in error state
+            let cardTitle = JSON.parse(error.responseText)?.errors?.[0]?.meta
+              ?.cardTitle;
+
+            if (!cardTitle) {
+              throw new Error(`Could not get card title for ${card}`);
+            }
 
-        const stackItem = here.allStackItems.find(
-          (item) => item.card === loadedCard,
-        );
-        // if is workspace index card, do not allow deletion
```

Comment:
We're not even providing a button to delete the card when the card is an index card, so this check is here is probably not needed.